src/context/cart/cart-context.js

```diff
@@ -0,0 +1,119 @@
+import { createContext, useContext, useReducer, useRef } from 'react';
+
+const initialCartState = {
+  cartProducts: [],
+  cartTotalItems: 0,
+  cartTotalPrice: 0,
```

Comment:
These two could be calculated from the `cartProducts` hence rather than having them as state variable you could just calculate them and use them where ever needed.

Reply:
I just wanted to keep the calculations at one place thats why i kept it here.